dynamic_stack_decider/dynamic_stack_decider/tree.py

```diff
@@ -28,7 +27,7 @@ class AbstractTreeElement:
     use one of DecisionTreeElement and ActionTreeElement instead
     """
 
-    def __init__(self, name, parent):
+    def __init__(self, name: str, parent: Optional["AbstractTreeElement"]):
```

Comment:
The type is not defined yet, as this is the constructor of the type itself. So it essentially says trust me bro, this will make sense later on. It is called forward reference. See this https://peps.python.org/pep-0484/#forward-references